cmd/fsck/main.go

```diff
@@ -0,0 +1,266 @@
+// Copyright 2025 The Tessera authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// fsck is a command-line tool for checking the integrity of a static-ct based log.
+package main
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"flag"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	tdnote "github.com/transparency-dev/formats/note"
+	"github.com/transparency-dev/merkle/rfc6962"
+	"github.com/transparency-dev/tessera/api/layout"
+	"github.com/transparency-dev/tessera/fsck"
+	"github.com/transparency-dev/tesseract/internal/client"
+	"golang.org/x/crypto/cryptobyte"
+	"golang.org/x/mod/sumdb/note"
+	"golang.org/x/sync/errgroup"
+	"k8s.io/klog/v2"
+)
+
+var (
+	storageURL  = flag.String("storage_url", "", "Base tlog-tiles URL")
+	bearerToken = flag.String("bearer_token", "", "The bearer token for authorizing HTTP requests to the storage URL, if needed")
+	N           = flag.Uint("N", 1, "The number of workers to use when fetching/comparing resources")
+	origin      = flag.String("origin", "", "Origin of the log to check")
+	pubKey      = flag.String("public_key", "", "The log's public key in base64 encoded DER format")
+)
+
+func main() {
+	klog.InitFlags(nil)
+	flag.Parse()
+	ctx := context.Background()
+	logURL, err := url.Parse(*storageURL)
+	if err != nil {
+		klog.Exitf("Invalid --storage_url %q: %v", *storageURL, err)
+	}
+	hc := &http.Client{
+		Transport: &http.Transport{
+			MaxIdleConns:        int(*N),
+			MaxIdleConnsPerHost: int(*N),
+			DisableKeepAlives:   false,
+		},
+		Timeout: 30 * time.Second,
+	}
+	src, err := client.NewHTTPFetcher(logURL, hc)
+	if err != nil {
+		klog.Exitf("Failed to create HTTP fetcher: %v", err)
+	}
+	if *bearerToken != "" {
+		src.SetAuthorizationHeader(fmt.Sprintf("Bearer %s", *bearerToken))
+	}
+	v := verifierFromFlags()
+	if *origin == "" {
+		*origin = v.Name()
```

Comment:
I think you can remove this check? `verifierFromFlags` will fail if origin is empty.

Reply:
Good spot, ta!